src/check.rs

```diff
@@ -94,42 +245,213 @@ fn check_http_url(url: &Url, ctx: &CheckContext) -> Result<(), CheckError> {
         }
     }
 
-    let resp = ureq::head(url.as_str()).call();
-    if resp.synthetic() {
-        Err(CheckError::Http(Box::new(HttpError::Fetch(
-            url.clone(),
-            resp.into_synthetic_error().unwrap(),
-        ))))
-    } else if resp.ok() {
-        Ok(())
+    // only if the URL contains a fragment we need to fetch the body via
+    // GET, otherwise we can use HEAD
+    if url.fragment().is_none() {
+        let resp = ureq::head(url.as_str()).call();
+
+        handle_response(url, resp).map(|_: ureq::Response| ())
```

Comment:
As a follow-up, not necessarily here, it might be nice to cache URLs that don't have fragments too.